### ISSUE TITLE
Error on 2.0.3: ssl_wrap_socket_with_ocsp() got an unexpected keyword argument 'key_password'

### ISSUE THREAD

**Issue by thalesmello** [snowflakedb/snowflake-connector-python#226]
Please answer these questions before submitting your issue. Thanks!

1. What version of Python are you using (`python --version`)? 3.7.2

2. What operating system and processor architecture are you using (`python -c 'import platform; print(platform.platform())'`)? Mac OS X

3. What are the component versions in the environment (`pip list`)? 
https://termbin.com/28i3

4. What did you do?
If possible, provide a recipe for reproducing the error.
A complete runnable program is good.
Install dbt with 0.10.3 and snowflake-python-connect 2.0.3.
`dbt debug` on any dbt project that uses snowflake and it will show

ssl_wrap_socket_with_ocsp() got an unexpected keyword argument 'key_password'

5. What did you expect to see?
No error

6. What did you see instead?'
ssl_wrap_socket_with_ocsp() got an unexpected keyword argument 'key_password'

7. Can you set logging to DEBUG and collect the logs?

```
import logging
import os

for logger_name in ['snowflake.sqlalchemy', 'snowflake.connector', 'botocore']: 
    logger = logging.getLogger(logger_name)
    logger.setLevel(logging.DEBUG)
    ch = logging.StreamHandler()
    ch.setLevel(logging.DEBUG)
    ch.setFormatter(logging.Formatter('%(asctime)s - %(threadName)s %(filename)s:%(lineno)d - %(funcName)s() - %(levelname)s - %(message)s'))
    logger.addHandler(ch)
```

**Comment by smtakeda** [comment]
The latest Snowflake connector depends on urllib3:
https://github.com/snowflakedb/snowflake-connector-python/blob/master/setup.py#L165

But your pip list doesn't include it.

Can you install an appropriate version of `urllib3`?

**Comment by naga-kondapaturi-zefr** [comment]
Got the same issue. It seems the latest release is broken I had to pin mine to the previous version.
snowflake-connector-python==2.0.2

**Comment by jacob-indigo** [comment]
This is the problem: https://github.com/snowflakedb/snowflake-connector-python/blob/master/network.py#L78. Because this is happening in the global scope, it is changing urllib3 for **everything**. This needs to be removed. You could potentially move this to some context manager but even then it would be very unsafe to do.

**Comment by joshjohnlowe** [comment]
I'm having the same issue. 2.0.3 fixed the Azure issue. I've tried pinning urllib to 1.24.3, pinning azure-storage-blob to 2.1.0, and pinning snowflake-connector-python to older versions with no luck.

**Comment by naga-kondapaturi-zefr** [comment]
> I'm having the same issue. 2.0.3 fixed the Azure issue. I've tried pinning urllib to 1.24.3, pinning azure-storage-blob to 2.1.0, and pinning snowflake-connector-python to older versions with no luck.

I didn't change urllib. Just pinned these 2 versions.

azure-storage-blob==2.1.0
snowflake-connector-python==2.0.2

**Comment by dinedal** [comment]
Same issue, pinning to 2.0.1 or 2.0.2 resolves

**Comment by smtakeda** [comment]
My working pip list is this:
```
>pip list
Package                    Version  
-------------------------- ---------
asn1crypto                 1.2.0    
azure-common               1.1.23   
azure-storage-blob         2.1.0    
azure-storage-common       2.1.0    
boto3                      1.10.9   
botocore                   1.13.9   
certifi                    2019.9.11
cffi                       1.13.2   
chardet                    3.0.4    
cryptography               2.8      
docutils                   0.15.2   
future                     0.18.2   
idna                       2.8      
ijson                      2.5.1    
jmespath                   0.9.4    
oscrypto                   1.1.0    
pip                        19.3.1   
pycparser                  2.19     
pycryptodomex              3.9.1    
PyJWT                      1.7.1    
pyOpenSSL                  19.0.0   
python-dateutil            2.8.0    
pytz                       2019.3   
requests                   2.22.0   
s3transfer                 0.2.1    
setuptools                 41.1.0   
six                        1.12.0   
snowflake-connector-python 2.0.3    
urllib3                    1.24.3   
```

**Comment by mgouline** [comment]
Pinning urllib3 to what @smtakeda listed worked for me. Version 2.0.3 doesn't work with urllib3 versions newer than 1.24.x so I ended up running `pip install urllib3<1.25` _before_ installing snowflake-connector-python, boto3 and all the others (order matters because those request other versions of urllib3).

**Comment by snowstakeda** [comment]
v2.0.3 pins urllib3
https://github.com/snowflakedb/snowflake-connector-python/blob/master/setup.py#L165
Maybe because the latest `requests` demands newer `urllib3`?
But my case always works with ```pip install -U pip && pip install -U snowflake-connector-python```

**Comment by eedwards-sk** [comment]
Hi, why was this closed?

We can't pin requests just to accommodate `snowflake-connector-python`, and thus we're now broken on latest.

We also noticed that 1.9.0 was removed from PyPI without any notice, yet is still listed in the release notes. This broke our build process and forced us to upgrade. It's considered extremely dangerous to remove packages after they've been published, and the lack of transparency is even more unsettling.

This is extremely unprofessional for a product used in production.

**Comment by smtakeda** [comment]
@eedwards-sk, Sorry about removal of 1.9.0. That was our mistake and will not happen again.

This issue was closed because the issue can be addressed by upgrading all to the latest following snowflake-connector-python needs.

Now let me understand more preciously about your question. You said you cannot pin `requests` for other needs. Can you please tell which `requests` version you use?

We do our best to meet all needs but sometimes one solution break other use cases. The newer `requests` version is required to decouple the embedded requests in `botocore` in #195. But unless we know the cases, it is hard to predict and avoid the break. Your input is important.

**Comment by eedwards-sk** [comment]
>Sorry about removal of 1.9.0. That was our mistake and will not happen again.

That's simply not enough. You need to update the PyPI page. You need to declare that it was removed, and you need to describe why. It's unacceptable.

Last working version we had was 1.9.0 and you broke us when you removed it. Every version since then was broken with the Azure BlockBlobService error, which is only resolved with newest version `2.0.3`, which then has an absurd `urllib3` requirement, which requires us to downgrade `requests`.

It's a library. It is not appropriate to lock your transitive dependencies.

>This issue was closed because the issue can be addressed by upgrading all to the latest following snowflake-connector-python needs.

Incorrect.

We're awaiting a version that is compatible with the latest version of requests.

**Comment by snowstakeda** [comment]
ok, we'll update the doc and attempt to support the latest urllib3 and requests.

**Comment by keller00** [comment]
I have fixed the issue with the newest `urllib3` and will be pinning it much less aggressively for the next release.

**Comment by eedwards-sk** [comment]
@keller00 

Libraries can inform minimum constraints, but should generally not enforce maximums, leaving it up to the system administrator to lock versions if the library ends up not being compatible.

I'm hoping this is what you mean by "pinning it much less aggressively" :)

Thanks for the update!

**Comment by skypanther** [comment]
Let's hope it's fixed. I wasted a good hour yesterday trying to figure out why my Lambda stopped working only to find this issue and roll back the urllib3 version that adding the S3 connector lib had changed.

**Comment by sburns** [comment]
@skypanther what ended up working for you? My reqs right now look like

```
boto3<1.10.0
botocore<1.13.0
glom==19.10.0
requests
snowflake-sqlalchemy==1.1.16
snowflake-connector-python==2.0.2
```

but now I get this awful `ImportError("cannot import name 'BlockBlobService' from 'azure.storage.blob' (/var/task/azure/storage/blob/__init__.py)")` and I don't care at all about azure. Don't get me started as to why I have to install azure packages when running in AWS but that's for another day.

**Comment by skypanther** [comment]
My requirements are currently:

```
asn1crypto==1.2.0
astroid==2.3.2
azure-common==1.1.23
azure-storage-blob==2.1.0
azure-storage-common==2.1.0
boto3==1.10.1
botocore==1.12.253
certifi==2019.9.11
cffi==1.13.1
chardet==3.0.4
cryptography==2.8
docutils==0.15.2
future==0.18.1
idna==2.8
ijson==2.5.1
isort==4.3.21
jmespath==0.9.4
lazy-object-proxy==1.4.2
mccabe==0.6.1
oscrypto==1.1.0
pycparser==2.19
pycryptodomex==3.9.0
PyJWT==1.7.1
pylint==2.4.3
pyOpenSSL==19.0.0
python-dateutil==2.8.0
pytz==2019.3
requests==2.22.0
s3transfer==0.2.1
six==1.12.0
snowflake-connector-python
typed-ast==1.4.0
urllib3==1.24.3
wrapt==1.11.2
```

**Comment by lcorbel** [comment]
This issue seems to have been fixed by release 2.0.4.

**Comment by eedwards-sk** [comment]
The 2.0.4 release notes do not reflect anything related to this issue.

There is no milestone attached to this issue.

1.9.0 continues to be listed on the release notes with no mention of it being removed, or why.

2.0.4 already appears to have a py3.x issue as per above.

I'm waiting for a stable version, with an official acknowledgement of the version pinning fix, and then I will upgrade and report back.

**Comment by snowstakeda** [comment]
> The 2.0.4 release notes do not reflect anything related to this issue.
The 2.0.4 patched for workaround. More complete solution would be in 2.0.5

> There is no milestone attached to this issue.
Attached recently.

> 1.9.0 continues to be listed on the release notes with no mention of it being removed, or why.
v2.0.5 will have the doc update.

> 2.0.4 already appears to have a py3.x issue as per above.
Can you please elaborate this? We are not aware of py3 specific issue.

**Comment by eedwards-sk** [comment]
@snowstakeda Thanks for the update!

>Can you please elaborate this? We are not aware of py3 specific issue.

The issue is referenced immediately above my comment (#218).

I will keep an eye out for 2.0.5.

**Comment by matt-fleming** [comment]
While waiting for 2.0.5, I've resorted to the following requirements when running in a lambda.
`azure-common<2.0.0
azure-storage-blob<12.0.0
requests==2.21.0
urllib3==1.24.3
snowflake-connector-python==1.9.1`

**Comment by snowstakeda** [comment]
Decided to skip 2.0.5 and release 2.1.0 in the first week of December.

**Comment by cedricwebisoft** [comment]
I have the same error on production and had to downgrade requests, urllib3 and snowflake-connector-python.

**Comment by snowstakeda** [comment]
@cedricwebisoft this should be fixed in v2.1.0. Can you please try?

**Comment by snowstakeda** [comment]
v2.1.0 was released. If anybody has an problem, please file a new issue. Closing.
